### ISSUE TITLE
teamId in request is empty for interactive messages in direct message channels

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#10688]
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/MM-15271)

When using an [Interactive Message](https://docs.mattermost.com/developer/interactive-messages.html) in a direct message channel, the `teamId` field in the request is empty.

Propose we add the the id of the team that the user is currently viewing via `model.PostActionIntegrationRequest`.

**Comment by carmo-evan** [comment]
I'd like to do this one.

**Comment by hanzei** [comment]
Wow, you are working on a lot of issue. Thanks for this @carmo-evan! :+1:

**Comment by carmo-evan** [comment]
@hanzei @jasonblais the conversation about this has stalled [here](https://community.mattermost.com/core/pl/y5rkx45ccbfqbno15zaq9uxwhy)

**Comment by jasonblais** [comment]
@hanzei what would be your thoughts on Jesse's response at https://community.mattermost.com/core/pl/y5rkx45ccbfqbno15zaq9uxwhy ?

**Comment by hanzei** [comment]
I've responded in the thread. Sorry for the delay.

**Comment by hanzei** [comment]
This issue with be fixed in a different way. See https://github.com/mattermost/mattermost-webapp/pull/3313 for more details.

**Comment by dansmithy** [comment]
Is this fixed in Mattermost 5.18.0 ?

I'm seeing unexpected behavior that *might* be caused by the issue described here: interactive buttons/menus do not work in a direct message. 

Using Chrome dev tools, I can see the webapp post to `http://localhost:8065/api/v4/posts/zcrco8hbppbutgyazoahngfgca/actions/tiprgoyh53fj3edmfi6q3thkye` and get back a response like this:

```
{"id":"store.sql_team.get.find.app_error","message":"Unable to find the existing team","detailed_error":"","request_id":"s3tbgf44cfb7dyszioobwa63yh","status_code":404}
```

I would expect that either:

1) Interactive messages are *not* supported in direct messages, and I shouldn't be able to post one there.
2) Interactive messages *are* supported in direct messages, and when clicking a button my registered integration endpoint should receive a payload (with or without a team id).

Am I missing something? How do I get an issue raised for this? Thanks.

**Comment by jasonblais** [comment]
@dansmithy Are you open to creating a new GitHub issue for your issue report? Feel free to reference this issue (#10688) in your report.

cc @hanzei FYI
